colossalai/utils/profiler/common.py

```diff
@@ -0,0 +1,75 @@
+from enum import Enum
+from .comm_profiler import CommProfiler
+from typing import List
+from torch.utils.tensorboard import SummaryWriter
+
+
+class ProfilerEvent(Enum):
+    '''
+    Event types to profile
+    '''
+    COMM = CommProfiler
+
+
+class ProfilerContext(object):
+
+    def __init__(self, events: List[ProfilerEvent]):
+        self.events = list()
+        for event_cls in events:
+            self.events.append(event_cls())
+
+    def __enter__(self):
+        for e in self.events:
+            e.enable()
+
+    def __exit__(self):
+        for e in self.events:
+            e.disable()
+
+    def to_tensorboard(self, writer: SummaryWriter):
+        for e in self.events:
+            e.to_tensorboard(writer)
+
+    def to_file(self, filename: str):
+        for e in self.events:
+            e.to_file(filename)
+
+    def show(self):
+        for e in self.events:
+            e.show()
+
+    def get_last(self):
+        res = dict()
+        for e in self.events:
+            event_result = e.get_last()
+            if event_result is not None:
+                for k, v in event_result.items():
+                    res[k] = v
+        return res
+
+    def get_avg(self):
+        res = dict()
+        for e in self.events:
+            event_result = e.get_avg()
+            if event_result is not None:
+                for k, v in event_result.items():
+                    res[k] = v
+        return res
+
+
+def colossalai_profiler(*args, **kwargs):
+    '''
+    Profiler context manager
+
+    Usage:
+    from colossalai.utils.profiler import ProfilerEvent, colossalai_profiler
+    from torch.utils.tensorboard import SummaryWriter
+
+    with colossalai_profiler(events=[ProfilerEvent.COMM]) as prof:
+        train()
+    writer = SummaryWriter('tb/path')
+    prof.to_tensorboard(writer)
+    prof.to_file('./prof.log')
+    prof.show()
+    '''
+    return ProfilerContext(*args, **kwargs)

```

Comment:
This wrapper does not do anything, can just let user import ProfilerContext

Reply:
Just mean to use lowercase api for context

Reply:
It is alright to use `ProfilerContext`. In fact, the `*args` and `**kwargs` can hurt usability.